Dockerfile

```
@@ -1,11 +1,17 @@
+# check=skip=FromPlatformFlagConstDisallowed
+# Prod and CI are x86_64; pin amd64 so local (e.g. Apple Silicon) builds produce
+# a deployable artifact instead of an arm64 image the droplet can't run.
+FROM --platform=linux/amd64 ubuntu:latest AS builder
 
 ARG BUILD_PROFILE=release
 
 RUN apt update -y
 RUN apt install curl git -y
+# filter-syscalls is disabled because QEMU-emulated amd64 builds (e.g. locally on
+# Apple Silicon) can't load the seccomp BPF program, which fails the installer.
 RUN curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install linux \
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)